Mackbellemore/soen-390-team07
# refactor(client): consolidate react router routes and 404 page

#

## Changes this PR introduces

- The router configs are now consolidated in the `constants.js` file, making it easier to make future modifications
- The components related to the sidebar are better organized
- The contents of the sidebar render depending on the currently logged-in user's role
- Each path in the route are lazy loaded for better initial loading times
- Add a 404 page
- Reduce bundle size of antd



## How to test this PR

- Browse around the app and witness the lazy-loading
- Mistype a URL and see the 404 page
- Log in as a general user and you should not see the Admin item in the sidebar











## Checklist
